### PR TITLE
Add device add/delete to inventory

### PR DESCRIPTION
**Pull request by cheynearista**
- add_device_to_inventory
- retry_add_to_inventory
- delete_device
- delete_devices
- get_non_connected_device_count
- save_inventory

### CODE REVIEW

File: cvprac/cvp_api.py [hunk 1]
```diff
@@ -249,6 +249,87 @@ def get_inventory(self, start=0, end=0, query=''):
                              (query, start, end), timeout=self.request_timeout)
         return data['netElementList']
 
+    def add_device_to_inventory(self, device_ip, parent_name, parent_key):
+        ''' Add the container to the specified parent.
```
Comment (mharista) [inline cvprac/cvp_api.py]: Should this say "Add the device to the specified parent container." ?

File: cvprac/cvp_api.py [hunk 2]
```diff
@@ -249,6 +249,87 @@ def get_inventory(self, start=0, end=0, query=''):
...
+                        {
+                             'containerName' : parent_name,
+                             'containerId' : parent_key,
+                             'containerType' : 'Existing',
+                             'ipAddress' : device_ip,
+                             'containerList' : []
+                        }
+                        ]
+                }
+        data = self.clnt.post('/inventory/add/addToInventory.do?'
```
Comment (mharista) [inline cvprac/cvp_api.py]: Is intention to return something from or all of data?

File: cvprac/cvp_api.py [hunk 3]
```diff
@@ -249,6 +249,87 @@ def get_inventory(self, start=0, end=0, query=''):
...
+                username (str): username for device login
+                password (str): password for user
+        '''
+        self.log.debug('retry_add_to_inventory: called')
+        data = { "key" : device_mac,
+                 "ipAddress" : device_ip,
+                 "userName" : username,
+                 "password" : password
+               }
+        data = self.clnt.post('/inventory/add/retryAddDeviceToInventory.do?'
```
Comment (mharista) [inline cvprac/cvp_api.py]: Is intention to return something from or all of data?

File: cvprac/cvp_api.py [hunk 4]
```diff
@@ -249,6 +249,87 @@ def get_inventory(self, start=0, end=0, query=''):
...
+                device_macs (list): list of mac address for 
+                                    devices we're deleting
+        '''
+        self.log.debug('delete_devices: called')
+        data = {'data': device_macs}
+        return self.clnt.post('/inventory/deleteDevices.do?',
+                               data=data,
+                               timeout=self.request_timeout)
+
+    def get_non_connected_device_count(self):
```
Comment (mharista) [inline cvprac/cvp_api.py]: Is this devices that are unreachable or devices that are connected and not yet added to inventory?

Comment (cheynearista) [reply cvprac/cvp_api.py]: Devices that are added to the temp Inventory but are currently connected. This gets called before saving the inventory. I'll add more to the description.

File: cvprac/cvp_api.py (hunk 3 above)
Comment (cheynearista) [reply cvprac/cvp_api.py]: Should return nothing. I'll fix.

File: cvprac/cvp_api.py (hunk 2 above)
Comment (cheynearista) [reply cvprac/cvp_api.py]: Should return nothing. I'll fix.

File: cvprac/cvp_api.py (hunk 1 above)
Comment (cheynearista) [reply cvprac/cvp_api.py]: Fixed this.

File: test/system/test_cvp_api.py [hunk 5]
```diff
@@ -656,6 +658,38 @@ def test_api_apply_remove_image_container(self):
...
+        # delete from inventory
+        #self.api.delete_device(device['systemMacAddress'])
+        # verify not found in inventory
+        #res = self.api.get_device_by_name(device['fqdn'])
+        #self.assertEqual(res, {})
+        #dut = self.duts[0]
+        #self.api.retry_add_to_inventory(device['ipAddress'], 
+        #                                device['systemMacAddress'],
+        #                                dut['username'], dut['password'])
+
```
Comment (mharista) [inline test/system/test_cvp_api.py]: Should this be uncommented or removed?

File: test/system/test_cvp_api.py [hunk 6]
```diff
@@ -656,6 +658,38 @@ def test_api_apply_remove_image_container(self):
...
+        
+        # delete from inventory
+        #self.api.delete_device(device['systemMacAddress'])
+        # verify not found in inventory
+        #res = self.api.get_device_by_name(device['fqdn'])
+        #self.assertEqual(res, {})
+        #dut = self.duts[0]
+        #self.api.retry_add_to_inventory(device['ipAddress'], 
+        #                                device['systemMacAddress'],
+        #                                dut['username'], dut['password'])
```
Comment (mharista) [inline test/system/test_cvp_api.py]: Should this be uncommented or removed?

**Pull request merged**